yarn-project/archiver/src/archiver/validation.ts

```diff
@@ -0,0 +1,61 @@
+import type { EpochCache } from '@aztec/epoch-cache';
+import type { Logger } from '@aztec/foundation/log';
+import { type PublishedL2Block, getAttestationsFromPublishedL2Block } from '@aztec/stdlib/block';
+import { type L1RollupConstants, getEpochAtSlot } from '@aztec/stdlib/epoch-helpers';
+
+/**
+ * Validates the attestations submitted for the given block.
+ * Returns true if the attestations are valid and sufficient, false otherwise.
+ */
+export async function validateBlockAttestations(
+  publishedBlock: Pick<PublishedL2Block, 'attestations' | 'block'>,
+  epochCache: EpochCache,
+  constants: Pick<L1RollupConstants, 'epochDuration'>,
+  logger?: Logger,
+): Promise<boolean> {
+  const attestations = getAttestationsFromPublishedL2Block(publishedBlock);
+  const { block } = publishedBlock;
+  const blockHash = await block.hash().then(hash => hash.toString());
+  const archiveRoot = block.archive.root.toString();
+  const slot = block.header.getSlot();
+  const epoch = getEpochAtSlot(slot, constants);
+  const { committee } = await epochCache.getCommitteeForEpoch(epoch);
+  const logData = { blockNumber: block.number, slot, epoch, blockHash, archiveRoot };
+  logger?.debug(`Validating attestations for block ${block.number} at slot ${slot} in epoch ${epoch}`, {
+    committee: (committee ?? []).map(member => member.toString()),
+    recoveredAttestors: attestations.map(a => a.getSender().toString()),
+    postedAttestations: publishedBlock.attestations.map(a =>
+      a.address.isZero() ? a.signature.toString() : a.address.toString(),
+    ),
+    ...logData,
+  });
+
+  if (!committee || committee.length === 0) {
+    // Q: Should we accept blocks with no committee?
+    logger?.warn(`No committee found for epoch ${epoch} at slot ${slot}. Accepting block without validation.`, logData);
+    return true;
+  }
+
+  const committeeSet = new Set(committee.map(member => member.toString()));
+  const requiredAttestationCount = Math.floor((committee.length * 2) / 3) + 1;
```

Comment:
Is this 2 thirds configured on the contract somewhere? Should we read it rather than hard-code?

Reply:
It's hardcoded in the contract as well